internal/mode/static/state/graph/nginxproxy.go

```diff
@@ -40,12 +54,12 @@ func gcReferencesAnyNginxProxy(gc *v1.GatewayClass) bool {
 // validateNginxProxy performs re-validation on string values in the case of CRD validation failure.
 func validateNginxProxy(
```

Comment:
would it make sense to change getNginxProxy i nginxproxy.go to buildNginxPrxy, where we call validateNginxProxy(...) to determine if NginxProxy is valid. This way, `validateNginxProxy` will not have any side effects. 

```
func buildNginxProxy(...) *NginxProxy {
   valid = validateNginxProxy(...)
   return &NginxProxy{Source: ..., Valid: valid}
}
```

this will also simplify validateGatewayClass where we will no longer need to call validateNginxProxy.

I think we store ErrorMsg in NginxProxy struct, so that we can use it in validateGatewayClass